src/gui/PageView.cpp

```diff
@@ -371,6 +371,33 @@ auto XojPageView::onButtonPressEvent(const PositionInputData& pos) -> bool {
     return true;
 }
 
+auto XojPageView::onButtonClickEvent(const PositionInputData& pos) -> bool {
+    Control* control = xournal->getControl();
+    double x = pos.x;
+    double y = pos.y;
+
+    if (x < 0 || y < 0) {
+        return false;
+    }
+
+    double zoom = xournal->getZoom();
+    x /= zoom;
+    y /= zoom;
+
+    ToolHandler* h = control->getToolHandler();
+
+    if (h->getToolType() == TOOL_FLOATING_TOOLBOX) {
+        gint wx = 0, wy = 0;
+        GtkWidget* widget = xournal->getWidget();
+        gtk_widget_translate_coordinates(widget, gtk_widget_get_toplevel(widget), 0, 0, &wx, &wy);
+
+        wx += std::lround(pos.x) + this->getX();
+        wy += std::lround(pos.y) + this->getY();
+
+        control->getWindow()->floatingToolbox->show(wx, wy);
+    }
```

Comment:
This code is the same as code in `XojPageView::onButtonPressEvent` (and also `XojPageView::onButtonReleaseEvent`). I think it makes sense to add a private function to `PageView.cpp`:
```c++
/**
 * Shows the floating toolbox at the location of an input event.
 */
static void showFloatingToolbox(XournalView* xournal, const PositionInputData& pos) {
    Control* control = xournal->getControl();

    gint wx = 0, wy = 0;
    GtkWidget* widget = xournal->getWidget();
    gtk_widget_translate_coordinates(widget, gtk_widget_get_toplevel(widget), 0, 0, &wx, &wy);

    wx += std::lround(pos.x) + this->getX();
    wy += std::lround(pos.y) + this->getY();

    control->getWindow()->floatingToolbox->show(wx, wy);
}
```

...that we can use to reduce code duplication!